js/heatmap.js

```diff
@@ -119,6 +115,55 @@ window.addEventListener('DOMContentLoaded', () => {
       keys.forEach((keyName) => { keyFinger[keyName] = f; });
     });
 
+    const requiresExtension = keyCode => (
+      Array.from("TGBNHY").some(l => l == keyCode.at(3)) ||  // IndexInner
+      !(  // Not in 3×10 matrix
+        keyCode.startsWith('Key')
+        || ['Space', 'Comma', 'Period', 'Slash', 'Semicolon'].includes(keyCode)
+      )
+    );
+
+    const getKeyRow = keyCode => {
+      if (keyCode == "Space") return 0;
+      if (keyCode.startsWith('Digit')) return 4;
+
+      if (keyCode.startsWith('Key')) {
+        const letter = keyCode.at(3);
+        if (Array.from("QWERTYUIOP").some(l => l == letter)) return 3;
+        if (Array.from("ASDFGHJKL").some(l => l == letter)) return 2;
+        if (Array.from("ZXCVBNM").some(l => l == letter)) return 1;
```

Comment:
`.indexOf(letter) >= 0` would work as well (and be a bit quicker), but that’s a matter of preferences